### ISSUE TITLE
MKV extracting doesn't work

### ISSUE THREAD

**Issue by tivervac** [pannal/Sub-Zero.bundle#758]
Some basic info:
* Running Plex 1.21.2.3943 on a Synology DS418 NAS
* Manually installed Sub-Zero 2.6.5.3247 and made sure to `chown` the bundle to my plex user and group
* Kitana 0.4.0 running on my local macbook pro connected to the above plex, all on the same local network
* All PMS and Sub-Zero logging enabled

What I want to achieve is very simple, I just want to use Sub-Zero to extract the ASS subtitles from my .mkv video files and convert them to .srt.

After installation I made sure to place Sub-Zero as the first agent, followed by Local Media Assets. The library I'm experimenting is a `TV Shows` library that uses the `TheTVDB` agent.

I've set the following settings for Sub-Zero
* Download subtitles: never
* Embedded streams: Treat "Undefined" (und) as language 1: on
* All providers disabled
* Automatically extract and use embedded subtitles upon media addition (with configured default mods): on
* Don't search for subtitles of a language if there are embedded subtitles inside the media file (MKV/MP4)?: on
* Store subtitles next to media files (instead of metadata): on
* Subtitle formats to save: SRT

The first oddity I encounter is that Sub-Zero is not showing up in the plugins tab and that Kitana says  `No viable plugins found; please choose a different server`.

Normally I would say that means the plugin isn't running, however the logs say otherwise. OTOH, having scoured through the issues a bit, I've found that Sub-Zero is supposed to log a message when it's extracting a sub from a video file. I do not see such a log line.

Below you can find the log of stopping plex, starting it again and then pressing `Refresh metadata` on an episode that has embedded subtitles. That last step is repeated again after 30 seconds or so.

Any help would be appreciated.

[plex.log](https://github.com/pannal/Sub-Zero.bundle/files/5915378/plex.log)

**Comment by pannal** [comment]
> The first oddity I encounter is that Sub-Zero is not showing up in the plugins tab and that Kitana says No viable plugins found; please choose a different server.

> Normally I would say that means the plugin isn't running, however the logs say otherwise. OTOH, having scoured through the issues a bit, I've found that Sub-Zero is supposed to log a message when it's extracting a sub from a video file. I do not see such a log line.

Which logs say otherwise? Do you have subzero logs that show that it's actually being called/is running?

**Comment by tivervac** [comment]
@pannal I can not see any sub-zero logs which makes me think it's not working. However, the plex.log linked above has the following lines which make me think that sub-zero is doing something

> Feb 03, 2021 01:43:47.785 [0x7f922b31a0] Debug — Request: [127.0.0.1:44660 (Loopback)] GET /system/messaging/clear_events/com.plexapp.agents.subzero (12 live) GZIP Signed-in Token (XXXXXXX)
>Feb 03, 2021 01:43:48.019 [0x7fa09241a0] Debug — Completed: [127.0.0.1:44660] 200 GET /system/messaging/clear_events/com.plexapp.agents.subzero (12 live) GZIP 234ms 280 bytes
>Feb 03, 2021 01:44:15.644 [0x7f908551a0] Debug — [com.plexapp.agents.subzero] Plug-in running on port 40720.
>Feb 03, 2021 01:44:15.644 [0x7f908551a0] Debug — Read configuration for [com.plexapp.agents.subzero], had 0 prefixes
>Feb 03, 2021 01:44:20.706 [0x7f916fb1a0] Debug — [com.plexapp.agents.subzero] Sending command over HTTP (GET): /:/plugins/com.plexapp.agents.subzero/messaging/function/X0FnZW50S2l0OlNlYXJjaA__/Y2VyZWFsMQoxCmxpc3QKMApyMAo_/Y2VyZWFsMQoyCmRpY3QKZGljdAo2CnMyCmVuczQKbGFuZ2IwczYKbWFudWFsYjBzNwpwcmltYXJ5aTAKczcKdmVyc2lvbnIxCnM2Cmt3YXJnc3M3ClRWX1Nob3dzMTAKbWVkaWFfdHlwZTUKczQKOTIzNHM4CnBhcmVudElEczQ3CmNvbS5wbGV4YXBwLmFnZW50cy50aGV0dmRiOi8vMjU5NjQwLzQvMT9sYW5nPWVuczQKZ3VpZGIxczUKZm9yY2VzMjYKY29tLnBsZXhhcHAuYWdlbnRzLnRoZXR2ZGJzMTMKcHJpbWFyeV9hZ2VudHM0CjkyMzVzMgppZHIwCg__

**Comment by pannal** [comment]
> The library I'm experimenting is a TV Shows library that uses the TheTVDB agent.

Did you explicitly enable SZ for TheTVDB? And if you can see SZ in agents, you _have_ a subzero log in `Application Support\Plex Media Server\Logs\PMS Plugin Logs`. It's pretty much impossible to have one and not the other.

**Comment by tivervac** [comment]
> Did you explicitly enable SZ for TheTVDB? And if you can see SZ in agents, you _have_ a subzero log in `Application Support\Plex Media Server\Logs\PMS Plugin Logs`. It's pretty much impossible to have one and not the other.

You're right, my bad. I do have a subzero log but it's not telling me too much.

You can see the requested settings and the log below.
At `12:07:57` I did a `Refresh metadata` on a series. The only line that really seems interesting is the following

> 2021-03-09 12:07:57,885 (7f9497e1f0) :  DEBUG (__init__:125) - Skipping Sub-Zero agent(s)

[subzero.log](https://github.com/pannal/Sub-Zero.bundle/files/6108038/subzero.log)

![Agents](https://user-images.githubusercontent.com/3389524/110461883-4f7bd300-80d0-11eb-92d1-fda52210f545.png)
![AnimeAgent](https://user-images.githubusercontent.com/3389524/110461885-50ad0000-80d0-11eb-8ff4-47bd39eed8bd.png)

**Comment by pannal** [comment]
Yeah that's weird. Can you refresh the metadata of an episode in question and post the logs a couple of minutes afterwards?

SZ doesn't seem to be called.

**Comment by tivervac** [comment]
Alright, according to the log I refreshed the metadata at `12:19:31` (it was 12:21 on my local, non plex related machine) I then pulled the logs 10 minutes later. I've also included the Server log that's mentioned in the SubZero log

[com.plexapp.agents.subzero.log](https://github.com/pannal/Sub-Zero.bundle/files/6414372/com.plexapp.agents.subzero.log)
[Plex Media Server.log](https://github.com/pannal/Sub-Zero.bundle/files/6414373/Plex.Media.Server.log)
